test/framework/src/main/java/org/opensearch/test/OpenSearchIntegTestCase.java

```diff
@@ -2349,4 +2358,88 @@ protected ClusterState getClusterState() {
         return client(internalCluster().getClusterManagerName()).admin().cluster().prepareState().get().getState();
     }
 
+    /**
+     * Refreshes all indices in the cluster when forceRefresh set to true and
+     * checks if replica shards caught up with primary shard when Segment Replication is enabled.
+     */
+    protected void waitForReplication(boolean forceRefresh, String... indices) {
+        if (forceRefresh) {
+            refresh(indices);
+        }
+        if (isInternalCluster()) {
+            if (indices.length == 0) {
+                indices = getClusterState().routingTable().indicesRouting().keySet().toArray(String[]::new);
+            }
+            try {
+                for (String index : indices) {
+                    IndexRoutingTable indexRoutingTable = getClusterState().routingTable().index(index);
+                    if (indexRoutingTable != null) {
+                        assertBusy(() -> {
+                            for (IndexShardRoutingTable shardRoutingTable : indexRoutingTable) {
+                                final ShardRouting primaryRouting = shardRoutingTable.primaryShard();
+                                if (primaryRouting.state().toString().equals("STARTED")) {
+                                    final String indexName = primaryRouting.getIndexName();
+                                    if (isSegmentReplicationEnabledForIndex(indexName)) {
+                                        final List<ShardRouting> replicaRouting = shardRoutingTable.replicaShards();
+                                        final IndexShard primaryShard = getIndexShard(primaryRouting, indexName);
+                                        final Map<String, StoreFileMetadata> primarySegmentMetadata = primaryShard.getSegmentMetadataMap();
+                                        for (ShardRouting replica : replicaRouting) {
+                                            if (replica.state().toString().equals("STARTED")) {
```

Comment:
The intent of this method is now a bit different than when it was `verifyStoreContent` - "waitForReplication" means we will wait for replicas to be current which with checks like this is not entirely true as we treat non active replicas as current and proceed.  I think to be considered accurate this needs to wait for the shard to be started.

Reply:
Yes, I will add some method documentation to clarify the usage of this method waitForReplication(). My understanding is we should only care about shards that are already started. I don't think it's useful to add some waiting behaviour until inactive shards get started. I think this should be done on the test calling this method itself. 

The intent of waitForReplication() should be only to wait until active/started replicas to catch up with primaries and not worry about inactive/non-started replicas.